packages/ra-core/src/form/FormDataConsumer.js

```diff
@@ -42,8 +43,58 @@ const REDUX_FORM_NAME = 'record-form';
  *     </Edit>
  * );
  */
-const FormDataConsumer = ({ children, formData, ...rest }) => {
-    const ret = children({ formData, ...rest });
+const FormDataConsumer = ({ children, formData, source, ...rest }) => {
+    let scopedFormData = formData;
+    let getSource;
+    let getSourceHasBeenCalled = false;
+
+    // If we have a source, we are in an ArrayInput component
```

Comment:
I'm not very confident about this assumption. It seems an indirect way of detecting that we are in an `ArrayInput` component, based on the current `SimpleFormIterator` implementation. Isn't there a more robust way to detect it?

Reply:
Like passing an optional prop?